src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,75 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public static final int FIGURE_COUNT = 5;
+    private Figure[] figures = {new Circle(), new IsoscelesTrapezoid(),
+            new Rectangle(), new RightTriangle(), new Square()};
+    private Figure figure;
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        figure = figures[random.nextInt(FIGURE_COUNT)];
+        getParametres(figure);
+        figure.setColor(colorSupplier.getRandomColor());
+        return figure;
+    }
+
+    public void getParametres(Figure figure) {
+        if (figure instanceof Circle) {
+            getCircle();
+        }
+        if (figure instanceof IsoscelesTrapezoid) {
+            getIsoscelesTrapezoid();
+        }
+        if (figure instanceof Rectangle) {
+            getRectangle();
+        }
+        if (figure instanceof RightTriangle) {
+            getRightTriangle();
+        }
+        if (figure instanceof Square) {
+            getSquare();
+        }
+    }
+
+    public Figure getCircle() {
+        double radius = random.nextInt(9) + 1;
+        figure = new Circle(radius);
+        return figure;
+    }
+
+    public Figure getIsoscelesTrapezoid() {
+        double lowerBaseSide = random.nextInt(9) + 1;
+        double upperBaseSide = random.nextInt(9) + 1;
+        double height = random.nextInt(9) + 1;
+        figure = new IsoscelesTrapezoid(lowerBaseSide, upperBaseSide, height);
```

Comment:
```suggestion
        Figure figure = new IsoscelesTrapezoid(lowerBaseSide, upperBaseSide, height);
```
аналогічно в інших методах